### PR TITLE
Tune the CI publishing procedure to the new artefact naming conventions

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -455,7 +455,7 @@ jobs:
         with:
           upload_url: ${{ steps.create_gh_release.outputs.upload_url }}
           asset_path: ./dist/target/scala3-${{ env.RELEASE_TAG }}.tar.gz
-          asset_name: scala-${{ env.RELEASE_TAG }}.tar.gz
+          asset_name: scala3-${{ env.RELEASE_TAG }}.tar.gz
```
Comment (smarter) [inline .github/workflows/ci.yaml]: Is this needed? scala-3.0.0-M1.tar.gz looks fine to me.

**Review by smarter — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -404,8 +404,8 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
           upload_url: ${{ steps.create_gh_release.outputs.upload_url }}
-          asset_path: ./dist/target/dotty-${{ env.RELEASE_TAG }}.tar.gz
-          asset_name: dotty-${{ env.RELEASE_TAG }}.tar.gz
+          asset_path: ./dist/target/scala3-${{ env.RELEASE_TAG }}.tar.gz
+          asset_name: scala-${{ env.RELEASE_TAG }}.tar.gz
```
Comment (griggt) [inline .github/workflows/ci.yaml:408]: Does this mean that the name of the tarball referenced in the uploaded `sha256sum.txt` file will be different from the name of the file posted for download?  i.e. the checksum file will say `scala3-3.0.0-M2.tar.gz` but the file will actually be `scala-3.0.0-M2.tar.gz`?

Comment (anatoliykmetyuk) [reply .github/workflows/ci.yaml:408]: Yes, exactly. This is a good reason to align names.

Comment (anatoliykmetyuk) [reply .github/workflows/ci.yaml]: Thanks for pointing out!

**Pull request merged**